message.go

```diff
@@ -1386,3 +1386,8 @@ func (m *prettyRedisMessage) MarshalJSON() ([]byte, error) {
 	}
 	return json.Marshal(obj)
 }
+
+// NewMockResult is a mock method to send back RedisResult with err for test coverage
+func NewMockResult(val RedisMessage, err error) RedisResult {
```

Comment:
I have mixed thoughts about this
Because even though this will provide us a way to cover roughly 20 more lines of code in ruediscompat command, we should not probably expose this as a part of rueidis package

But the only alternative to cover most of the other parts inside command would be to introduce interfaces instead of passing concrete structs and a large refactor. Otherwise, it seems difficult to achieve near-100% coverage.

What do you think? @rueian

Reply:
> we should not probably expose this as a part of rueidis package

Agreed. Actually, we don't need this function but we can use the same hack of the `mock` package instead:
https://github.com/redis/rueidis/blob/45cdf7eb946fed65e41ec78a654860d44b79a143/mock/result.go#L22-L25

Reply:
Oh wow
This is a great use of unsafePtrs